packages/block-editor/src/components/inserter/search-items.js

```diff
@@ -44,94 +43,98 @@ const removeMatchingTerms = ( unmatchedTerms, unprocessedTerms ) => {
 	);
 };
 
+export const searchBlockItems = (
+	items,
+	categories,
+	collections,
+	searchTerm
+) => {
+	const normalizedSearchTerms = normalizeSearchTerm( searchTerm );
+	if ( normalizedSearchTerms.length === 0 ) {
+		return items;
+	}
+
+	return searchItems( items, searchTerm, {
+		getCategory: ( item ) =>
+			find( categories, { slug: item.category } )?.title,
+		getCollection: ( item ) =>
+			collections[ item.name.split( '/' )[ 0 ] ]?.title,
+		getVariations: ( item ) =>
+			( item.variations || [] ).map( ( variation ) => variation.title ),
+	} ).map( ( item ) => {
+		if ( isEmpty( item.variations ) ) {
+			return item;
+		}
+
+		const matchedVariations = item.variations.filter( ( variation ) => {
+			return (
+				intersectionWith(
+					normalizedSearchTerms,
+					normalizeSearchTerm( variation.title ),
+					( termToMatch, labelTerm ) =>
+						labelTerm.includes( termToMatch )
+				).length > 0
+			);
+		} );
+		// When no patterns matched, fallback to all variations.
```

Comment:
I missed this comment when renaming patterns to variations, now it looks ever more confusing:
```suggestion
		// When no variations matched, fallback to all variations.
```